internal/decryption/aes_gcm.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2022 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package decryption
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+)
+
+func aesGCMDecrypt(log *EncryptedLogRecord, lek []byte) ([]byte, error) {
```

Comment:
If this method only works with an EncryptedLogRecord I'd recommend making it a method of EncryptedLogRecord instead

Reply:
changed the func signature to be more generic and moved GCM and CBC to a separate `aes` package